### PR TITLE
Add `subvortex` subnet and tests

### PR DESCRIPTION
**Pull request by basfroman**

### CODE REVIEW

**Review by thewhaleking — APPROVED** [review]
Interesting approach.

**Review by ibraheem-abe — APPROVED** [review]

**Comment by basfroman** [comment]
> Interesting approach.

yeah, I thought it was a working approach too. but ultimately i have some thoughts on how to rework the verification mechanism for `network` and `chain_endpoint` fields. I want to roll it out later when I have more free time. The logic will be the same, but more elegant and readable.

File: bittensor/core/subtensor.py [hunk 1]
```diff
@@ -685,16 +685,8 @@ def determine_chain_endpoint_and_network(
...
-                # Kiru Finney staging network.
-                return network, settings.FINNEY_ENTRYPOINT
-            elif network == "local":
-                return network, settings.LOCAL_ENTRYPOINT
-            elif network == "test":
-                return network, settings.FINNEY_TEST_ENTRYPOINT
-            elif network == "archive":
-                return network, settings.ARCHIVE_ENTRYPOINT
+        if network in settings.NETWORKS:
+            return network, settings.NETWORK_MAP[network]
```
Comment (ppolewicz) [inline bittensor/core/subtensor.py:689]:
```suggestion
        if chain_endpoint := settings.NETWORK_MAP.get(network):
            return network, chain_endpoint
```

File: bittensor/core/settings.py [hunk 2]
```diff
@@ -46,19 +46,14 @@
 FINNEY_TEST_ENTRYPOINT = "wss://test.finney.opentensor.ai:443/"
 ARCHIVE_ENTRYPOINT = "wss://archive.chain.opentensor.ai:443/"
 LOCAL_ENTRYPOINT = os.getenv("BT_SUBTENSOR_CHAIN_ENDPOINT") or "ws://127.0.0.1:9944"
+SUBVORTEX_ENTRYPOINT = "ws://subvortex.info:9944"
 
 NETWORK_MAP = {
     NETWORKS[0]: FINNEY_ENTRYPOINT,
     NETWORKS[1]: FINNEY_TEST_ENTRYPOINT,
     NETWORKS[2]: ARCHIVE_ENTRYPOINT,
     NETWORKS[3]: LOCAL_ENTRYPOINT,
```
Comment (ppolewicz) [inline bittensor/core/settings.py:55]:
```suggestion
NETWORK_MAP = {
    "finney": FINNEY_ENTRYPOINT,
    "test": FINNEY_TEST_ENTRYPOINT,
    "archive": ARCHIVE_ENTRYPOINT,
    "local": LOCAL_ENTRYPOINT,
    "subvortex": SUBVORTEX_ENTRYPOINT,
}
```
and later
```
NETWORKS = NETWORK_MAP.keys()
```
Normally I'd refactor the module to not use NETWORKS at all, but someone might be using it in subnet code, so lets leave it. One day when the interface is documented we can clean it up.

File: bittensor/core/subtensor.py (hunk 1 above)
Comment (basfroman) [reply bittensor/core/subtensor.py:689]: we check the `network` argument before `chain_endpoint` if defined

**Pull request merged**